src/sentry/api/endpoints/organization_events_root_cause_analysis.py

```diff
@@ -1,10 +1,71 @@
 from rest_framework.response import Response
+from snuba_sdk import Column, Function
 
 from sentry import features
 from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.organization_events import OrganizationEventsEndpointBase
+from sentry.search.events.builder import QueryBuilder
+from sentry.search.events.types import QueryBuilderConfig
+from sentry.search.utils import parse_datetime_string
+from sentry.snuba.dataset import Dataset
 from sentry.snuba.metrics_performance import query as metrics_query
+from sentry.utils.snuba import raw_snql_query
+
+
+def query_spans(transaction, regression_breakpoint, params):
+    selected_columns = [
+        "count(span_id) as span_count",
+        "sumArray(spans_exclusive_time) as total_span_self_time",
+        "array_join(spans_op) as span_op",
+        "array_join(spans_group) as span_group",
+        # want a single event id to fetch from nodestore for the span description
+        "any(id) as sample_event_id",
+    ]
+
+    builder = QueryBuilder(
```

Comment:
Just realized but you'll want to add something like a `limit by` clause otherwise spans from 1 period may dominate the list of results.

Take an extreme example where there are 10k unique spans and in the after, all 10k take longer, then your results will only contain after spans and no before spans.

Reply:
I decreased the limit to 50 for a smaller subset of spans and added a limit by on the `period` column with a value of `limit // 2`

https://github.com/getsentry/sentry/pull/56653/commits/29884f72a3895592eb73f06de3b0e047b6407a4e